src/pages/api/github/pr.ts

```diff
@@ -0,0 +1,139 @@
+import { Metadata } from "@/utils";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { getSession } from "next-auth/react";
+import { Octokit } from "@octokit/core";
+import slugify from "slugify";
+
+async function createForkAndPR(
+  octokit: InstanceType<typeof Octokit>,
+  directoryPath: string,
+  fileName: string,
+  transcribedText: string,
+  metaData: Metadata
+) {
+  const upstreamOwner = "bitcointranscripts";
+  const upstreamRepo = "bitcointranscripts";
+
+  const directoryName = directoryPath.split("/").slice(-1)[0];
+
+  // Fork the repository
+  const forkResult = await octokit.request("POST /repos/{owner}/{repo}/forks", {
+    owner: upstreamOwner,
+    repo: upstreamRepo,
+  });
+
+  const forkOwner = forkResult.data.owner.login;
+  const forkRepo = upstreamRepo;
+
+  // Get the ref for the base branch
+  // a gh/network delay might cause base branch not to be availabe so run in a retry function
+  // retry put on hold
+  const baseBranchName = forkResult.data.default_branch;
+  const baseBranch = await octokit.request(
+    "GET /repos/{owner}/{repo}/git/ref/{ref}",
+    {
+      owner: forkOwner,
+      repo: forkRepo,
+      ref: `heads/${baseBranchName}`,
+    }
+  );
+
+  // Create new branch
+  const timeInSeconds = Math.floor(Date.now() / 1000);
+  const newBranchName = `${timeInSeconds}-${directoryName}`;
+  const baseRefSha = baseBranch.data.object.sha;
+
+  await octokit.request("POST /repos/{owner}/{repo}/git/refs", {
+    owner: forkOwner,
+    repo: forkRepo,
+    ref: `refs/heads/${newBranchName}`,
+    sha: baseRefSha,
+  });
+
+  // Make edits to the file on the new branch
+  const metadata = metaData.toString();
+  const transcriptData = `${metadata}\n${transcribedText}\n`;
+  const fileContent = Buffer.from(transcriptData).toString("base64");
+
+  // Create new file on the branch
+  const path = directoryPath.split("/").slice(1).join("/");
+  const fileSlug = slugify(fileName, { remove: /[*+~.()'"!:@\\/]/g });
+  await octokit.request("PUT /repos/:owner/:repo/contents/:path", {
+    owner: forkOwner,
+    repo: forkRepo,
+    path: `${path}/${fileSlug}.md`,
```

Comment:
this does work for the How Bitcoin Works transcript.

The URL for the PUT is `https://api.github.com/repos/adamjonas/bitcointranscripts/contents/%2FHow-Bitcoin-Actually-Works-Episode-3-or-Hello-Bitcoin.md` and the error message is that ` message: 'path cannot start with a slash'`. There seems to be some UTF-8 encoding of `%2F`. I hard coded it for [#107](https://github.com/adamjonas/bitcointranscripts/pull/107/) to go through.

Reply:
So I had the wrong format for the location path. I accounted for `"bitcointranscripts/"` prepended to the location whereas tstbtc doesn't prepend it hence `${path}` didn't exist and `"/"` was appended to the path which is why it kept failing. I updated the logic in 2c80bf8bba046582a50281589e908e4aac63526c